File: 6KYU/Rectangle into Squares/solution.cs
using System.Collections.Generic;
public class SqInRect {
	
	public static List<int> sqInRect(int l, int w) {
		if(l == w) return null;
    var res = new List<int>();
    
    while(l != 0 && w != 0)
    {
      if(l < w) {
        res.Add(l);
        w -= l;
      } else {
        res.Add(w);
        l -= w;
      }
    }
    return res;
	}
}

File: 6KYU/Rectangle into Squares/solution.go
package kata

func SquaresInRect(l int, w int) []int {

	if l == w {
		return nil
	}

	res := []int{}
	for {
		if l == 0 || w == 0 {
			return res
		}

		if l < w {
			res = append(res, l)
			w -= l
		} else {
			res = append(res, w)
			l -= w
		}
	}
}
